src/externalized/DefaultContentManager.cc

```diff
@@ -181,8 +181,13 @@ const DealerInventory* DefaultContentManager::getBobbyRayUsedInventory() const
 	return m_bobbyRayUsedInventory.get();
 }
 
-const DealerModel* DefaultContentManager::getDealer(uint8_t dealerID) const
+const DealerModel* DefaultContentManager::getDealer(ArmsDealerID dealerID) const
 {
+	auto dealerIndex = static_cast<std::size_t>(dealerID);
+	if (dealerIndex >= m_dealers.size()) {
+		ST::string err = ST::format("failed to get dealer: invalid dealer ID: {}", dealerID);
+		throw std::runtime_error(err.to_std_string());
+	}
 	return m_dealers[dealerID];
```

Comment:
I wouldn't expect to see this exception too often (hopefully never) so the additional error info shouldn't be needed and the range checking could be handled by vector::at.